File: IN/sample.php
<?php
// -- CONSTANTS

const
    MINIMUM_PASS_COUNT = 0,
    MAXIMUM_PASS_COUNT = 5;

// -- TYPES

class AgeInterval
{
    // -- ATTRIBUTES

    public int
        firstAge,
        lastAge;

    // -- CONSTRUCTORS

    public function __construct(
        int $firstAge,
        int $lastAge
        )
    {
        $this->firstAge = $firstAge;
        $this->lastAge = $lastAge;
    }
}

// ~~

class Being
{
    // -- ATTRIBUTES

    public string
        $name;
    public int
        $age;

    // -- CONSTRUCTORS

    public function __construct(
        string $name,
        int $age
        )
    {
        $this->name = $name;
        $this->age = $age;
    }
}

// ~~

class Person extends Being
{
    // -- ATTRIBUTES

    public float
        $weight,
        $dogCount;

    // -- CONSTRUCTORS

    public function __construct(
        string $name,
        int $age,
        float $weight,
        float $dogCount
        )
    {
        parent::__construct( $name, $age );
        $this->weight = $weight;
        $this->dogCount = $dogCount;
    }

    // -- INQUIRIES

    public function getAge(
        ): int
    {
        return $this->age;
    }

    // ~~

    public function getAgeOffset(
        int $otherAge
        ): int
    {
        return $otherAge - $this->age;
    }

    // ~~

    public function getHelloMessage(
        ): string
    {
        return "Hello, my name is {$this->name}, I'm {$this->age} years old and I weight {$this->weight} kilograms.";
    }

    // -- OPERATIONS

    public function setAge(
        int $age
        ): void
    {
        $this->age = $age;
    }

    // ~~

    public function setFakeAge(
        int $age
        ): void
    {
        if ( $age > 0
             && $age < 50
             && ( $age < 20
                  || $age > 40 ) )
        {
            $this->age
                += ( $age
                     + ( $age - 2 )
                     + ( $age
                         * ( $age + 10 )
                         * ( $age + 20 ) ) )
                   + $this->getAgeOffset(
                        $age * 2
                        - 20
                        );
        }
        elseif ( $age > 20
                 && $age < 40
                 && ( $age < 25
                      || $age > 35 ) )
        {
            $this->age
                = (int)round(
                    $age * 0.5
                    );
        }
        else
        {
            $this->age = $age + 10;
        }
    }
}

// -- FUNCTIONS

function getAgeInterval(
    array $sortedPersonArray
    ): ?AgeInterval
{
    if ( count( $sortedPersonArray ) === 0 )
    {
        return null;
    }
    else
    {
        return
            new AgeInterval(
                $sortedPersonArray[ 0 ]->age,
                $sortedPersonArray[ count( $sortedPersonArray ) - 1 ]->age
                );
    }
}

// ~~

function main(
    ): void
{
    $passIndex = 0;

    while ( $passIndex < 5 )
    {
        $passIndex += 1;
    }

    while ( true )
    {
        $passIndex += 1;

        if ( $passIndex >= 10 )
        {
            break;
        }
    }

    $personArray
        = [
              new Person( 'Mike', 49, 85.0, 1.0 ),
              new Person( 'Luke', 30, 77.0, 0.0 ),
              new Person( 'John', 30, 72.0, 3.0 )
          ];

    usort(
        $personArray,
        function (
            Person $firstPerson,
            Person $secondPerson
            ): int
        {
            try
            {
                if ( $firstPerson->age !== $secondPerson->age )
                {
                    return $firstPerson->age <=> $secondPerson->age;
                }
                else
                {
                    return $firstPerson->weight <=> $secondPerson->weight;
                }
            }
            catch ( Throwable $error )
            {
                print( $error->getMessage() . PHP_EOL );
            }

            return 0;
        }
        );

    $ageInterval = getAgeInterval( $personArray );

    if ( $ageInterval !== null )
    {
        print( "First age: {$ageInterval->firstAge}" . PHP_EOL );
        print( "Last age: {$ageInterval->lastAge}" . PHP_EOL );
    }
    else
    {
        print( 'No age interval' . PHP_EOL );
    }
}

// -- STATEMENTS

main();

File: IN/sample.cs
// -- IMPORTS

using System;
using System.Collections.Generic;

// -- CONSTANTS

public static class SampleConstants
{
    public const int
        MinimumPassCount = 0,
        MaximumPassCount = 5;
}

// -- TYPES

public class Being
{
    // -- ATTRIBUTES

    public string
        Name;
    public int
        Age;

    // -- CONSTRUCTORS

    public Being(
        string name,
        int age
        )
    {
        Name = name;
        Age = age;
    }
}

// ~~

public class Person:
    Being
{
    // -- ATTRIBUTES

    public double
        Weight,
        DogCount;

    // -- CONSTRUCTORS

    public Person(
        string name,
        int age,
        double weight,
        double dogCount
        ):
        base( name, age )
    {
        Weight = weight;
        DogCount = dogCount;
    }

    // -- INQUIRIES

    public int GetAge(
        )
    {
        return Age;
    }

    // ~~

    public int GetAgeOffset(
        int otherAge
        )
    {
        return otherAge - Age;
    }

    // ~~

    public string GetHelloMessage(
        )
    {
        return $"Hello, my name is { Name }, I'm { Age } years old and I weight { Weight } kilograms.";
    }

    // -- OPERATIONS

    public void SetAge(
        int age
        )
    {
        Age = age;
    }

    // ~~

    public void SetFakeAge(
        int age
        )
    {
        if ( age > 0
             && age < 50
             && ( age < 20
                  || age > 40 ) )
        {
            Age
                += ( age
                     + ( age - 2 )
                     + ( age
                         * ( age + 10 )
                         * ( age + 20 ) ) )
                   + GetAgeOffset(
                         age * 2
                         - 20
                         );
        }
        else if ( age > 20
                  && age < 40
                  && ( age < 25
                       || age > 35 ) )
        {
            Age = (int)Math.Round( age * 0.5, MidpointRounding.AwayFromZero );
        }
        else
        {
            Age = age + 10;
        }
    }
}

// -- FUNCTIONS

public static class SampleFunctions
{
    public static Dictionary<string, int>? GetAgeInterval(
        List<Person> sortedPersonList
        )
    {
        if ( sortedPersonList.Count == 0 )
        {
            return null;
        }
        else
        {
            return new Dictionary<string, int>
            {
                { "firstAge", sortedPersonList[ 0 ].Age },
                { "lastAge", sortedPersonList[ sortedPersonList.Count - 1 ].Age }
            };
        }
    }
}

// ~~

public static class Program
{
    public static void Main(
        )
    {
        var passIndex = 0;

        while ( passIndex < 5 )
        {
            ++passIndex;
        }

        do
        {
            ++passIndex;
        }
        while ( passIndex < 10 );

        var personList
            = new List<Person>
            {
                new Person( "Mike", 49, 85, 1 ),
                new Person( "Luke", 30, 77, 0 ),
                new Person( "John", 30, 72, 3 )
            };

        personList.Sort(
            ( firstPerson, secondPerson ) =>
            {
                try
                {
                    if ( firstPerson.Age != secondPerson.Age )
                    {
                        return firstPerson.Age - secondPerson.Age;
                    }
                    else
                    {
                        return firstPerson.Weight.CompareTo( secondPerson.Weight );
                    }
                }
                catch ( Exception error )
                {
                    Console.WriteLine( error.ToString() );
                }

                return 0;
            }
            );

        var ageInterval = SampleFunctions.GetAgeInterval( personList );

        if ( ageInterval != null )
        {
            Console.WriteLine( $"First age: { ageInterval[ "firstAge" ] }" );
            Console.WriteLine( $"Last age: { ageInterval[ "lastAge" ] }" );
        }
        else
        {
            Console.WriteLine( "No age interval" );
        }
    }
}
